### PR TITLE
cryptonote: fix reuse of non default tx data when relaying

### PR DESCRIPTION
**Pull request by moneromooo-monero**
An automatic tx variable is initialized properly on the first
run through the loop, but not the second. Moving the variable
inside the loop ensures the ctor is called again to init it.

### CODE REVIEW

File: src/cryptonote_basic/cryptonote_format_utils.cpp [hunk 1]
```diff
@@ -146,6 +146,8 @@ namespace cryptonote
     if (tx.version >= 2 && !is_coinbase(tx))
     {
       rct::rctSig &rv = tx.rct_signatures;
+      if (rv.type == rct::RCTTypeNull)
```
Comment (vtnerd) [inline src/cryptonote_basic/cryptonote_format_utils.cpp:149]: Is this related?

File: src/cryptonote_core/cryptonote_core.cpp [hunk 2]
```diff
@@ -1308,9 +1308,9 @@ namespace cryptonote
     std::vector<crypto::hash> tx_hashes{};
     tx_hashes.resize(tx_blobs.size());
 
-    cryptonote::transaction tx{};
     for (std::size_t i = 0; i < tx_blobs.size(); ++i)
     {
+      cryptonote::transaction tx{};
```
Comment (vtnerd) [inline src/cryptonote_core/cryptonote_core.cpp:1313]: Crap. The internal memory gets freed every cycle, but there is no easier fix.

File: src/cryptonote_basic/cryptonote_format_utils.cpp (hunk 1 above)
Comment (moneromooo-monero) [reply src/cryptonote_basic/cryptonote_format_utils.cpp:149]: This is what made me find that bug in the first place. I guess it can be omitted since all null rct types are coinbase so will be caught by the previous check. But pedantically it's best to check the rct data exists before we get it below.

**Comment by dEBRUYNE-1** [comment]
@vtnerd - Can this be merged in the current state or do you have any further comments?
